eurovision/notebooks/story.ipynb

```diff
@@ -2090,7 +2039,269 @@
+    "    predictions_baseline = predict_baseline(rescaled_ranks, year)\n",
+    "    actual = get_actual_ranks(year)\n",
+    "    # Remove Slovakia from 1998\n",
+    "    if year == 1998:\n",
+    "        predictions_baseline = predictions_baseline.query('to_code2 != \"SK\"')\n",
+    "        actual = actual.query('to_code2 != \"SK\"')\n",
+    "    return(get_spearman(predictions_baseline, actual))\n",
+    "    \n",
+    "spearmans = [get_spearman_baseline_without(year) for year in all_years]\n",
+    "\n",
+    "print(f'Mean Spearman coefficient across {len(all_years)} years: {np.mean(spearmans)}')\n",
+    "\n",
+    "df_xv = pd.DataFrame({'year': all_years, 'spearman': spearmans})\n",
+    "fig, ax = plt.subplots(figsize=(6, 4))\n",
+    "ax.scatter(df_xv['year'], df_xv['spearman'], color='#6495ed')\n",
+    "ax.set(xlabel='Year', ylabel='Spearman coefficient', title='Cross-validation: baseline method')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It turns out that *on average*, the historical results of a country at Eurovision are a half-decent indicator of how well they will perform in any given year.\n",
+    "Of course, this is hardly a surprise.\n",
+    "Interestingly, we can pick out years with lower Spearman correlations as being 'anomalies' in this regard, where an underdog won.\n",
+    "The year with the lowest value (that's 2002) featured Latvia as the winner, and perhaps even more surprisingly, the UK finished third ([Jessica Garlick with *'Come Back'*](https://www.youtube.com/watch?v=2NSR5rSk0Cs)).\n",
+    "When we write 'surprisingly', this is emphatically *not* us inserting our personal opinion: the baseline model in fact really dislikes the UK, assigning it the third-lowest average rescaled rank of all countries.\n",
+    "Thus, any year where the UK performs well is a surprise to our model!\n",
+    "\n",
+    "Having looked at how good our model is at reproducing its own data, let's see what happens when we predict the 2023 results (now including all years' worth of data in our training)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictions_2023 = predict_baseline(train_baseline(), 2023)\n",
+    "\n",
+    "n = len(predictions_2023)\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(7, 7))\n",
+    "ax.barh(width=predictions_2023['rescaled_rank'], y=np.arange(n), color='#4878D0')\n",
+    "ax.set(\n",
+    "    xlim=(0, predictions_2023['rescaled_rank'].max() * 1.1),\n",
+    "    ylim=(-1, n),\n",
+    "    yticks=np.arange(n),\n",
+    "    yticklabels=predictions_2023.index.values,\n",
+    "    xlabel='Average rescaled rank across all final appearances',\n",
+    "    title='Baseline model: predicting 2023 results based on historical performance'\n",
+    ")\n",
+    "ax.invert_yaxis()\n",
+    "plt.show()"
+   ]
```

Comment:
This is a lovely graph, probably the one that people are going to be most interested in, so I'm wondering if it would make sense to map the two-letter codes to country names instead? The two-letter codes, even after all this time, still feel a bit opaque.

Reply:
<img width="726" alt="image" src="https://github.com/KatrionaGoldmann/Eurovision_TDS/assets/22414895/6fef5259-e718-42ce-9f2a-2429d9823259">

Reply:
Really nice; thanks.